src/branchandbound.jl

```diff
@@ -1,101 +1,59 @@
 # univariate
-function enclose_BranchandBound(f::Function, dom::Interval; order=10, tol=0.6)
-    x0 = Interval(mid(dom))
-    x = TaylorModel1(order, x0, dom)
-    return branchandbound(f(x-x0).pol, dom, tol)
-end
 
-# multivariate
-function enclose_BranchandBound(f::Function, dom::IntervalBox{D};
-                                order=10, tol=0.6) where {D}
-    x0 = mid(dom)
-    set_variables(Float64, "x", order=2order, numvars=D)
-    x = [TaylorModelN(i, order, IntervalBox(x0), dom) for i=1:D]
-    return branchandbound(f(x...).pol, dom - x0, tol)
+@inline function enclose(f::Function, X::Interval, ba::BranchAndBoundEnclosure;
+                         df=t->ForwardDiff.derivative(f, t))
+    return _branch_bound(ba, f, X, df)
 end
 
-@inline _Rnext(R::Vector{<:Interval}) = Interval(minimum(inf.(R)), maximum(sup.(R)))
+@inline function enclose(f::Function, X::IntervalBox, ba::BranchAndBoundEnclosure;
+                         df=t->ForwardDiff.gradient(w->f(w...), t.v))
```

Comment:
the `w->f(w...)` splatting is because of #75 , the `t.v` is because `IntervalBox` is not a subtype of `AbstractVector` and hence ForwardDiff doesn't work with it out of the box (pun not intended)

Reply:
I'll open an issue in interval arithmetic for the latter